src/components/SwitchLocaleLink/index.tsx

```diff
@@ -17,46 +14,35 @@ const Container = styled(TYPE.small)`
   margin-top: 1rem !important;
 `
 
-export function SwitchLocaleLink() {
-  const activeLocale = useActiveLocale()
+const useTargetLocale = (activeLocale: SupportedLocale) => {
   const browserLocale = useMemo(() => navigatorLocale(), [])
-  const location = useLocation()
-  const qs = useParsedQueryString()
 
   if (browserLocale && (browserLocale !== DEFAULT_LOCALE || activeLocale !== DEFAULT_LOCALE)) {
-    let targetLocale: SupportedLocale
     if (activeLocale === browserLocale) {
-      targetLocale = DEFAULT_LOCALE
+      return DEFAULT_LOCALE
     } else {
-      targetLocale = browserLocale
-    }
-
-    const target = {
-      ...location,
-      search: stringify({ ...qs, lng: targetLocale }),
+      return browserLocale
     }
-
-    return (
-      <Container>
-        <Trans>
-          Uniswap available in:{' '}
-          {
-            <StyledInternalLink
-              onClick={() => {
-                ReactGA.event({
-                  category: 'Localization',
-                  action: 'Switch Locale',
-                  label: `${activeLocale} -> ${targetLocale}`,
-                })
-              }}
-              to={target}
-            >
-              {LOCALE_LABEL[targetLocale]}
-            </StyledInternalLink>
-          }
-        </Trans>
-      </Container>
-    )
   }
   return null
 }
+
+export function SwitchLocaleLink() {
+  const activeLocale = useActiveLocale()
+  const targetLocale = useTargetLocale(activeLocale)
+
+  const { to, onClick } = useLocationLinkProps(targetLocale)
+
+  if (!targetLocale || !to) return null
+
+  return (
+    <Container>
+      <Trans>
+        Uniswap available in:{' '}
```

Comment:
@nenadV91 @W3stside , this is locale related, and also has "Uniswap" in it, so we will need to personalize

Reply:
it is already modded